src/util/alarm.cpp

```diff
@@ -0,0 +1,93 @@
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "../hal/timer.h"
+
+#include "../util/alarm.h"
+
+static struct alarm* alarmList = NULL;
+
+void alarmInit(struct alarm* alarm, const struct alarmProperties* properties)
+{
+  // Initialize all the parts of the alarm
+  alarm->set = false;
+  alarm->suppressed = false;
+  alarm->properties = properties;
+  alarm->next = NULL;
+  
+  // Insert into the list of all alarms in the system
+  struct alarm** current = &alarmList;
+  while (*current) {
+    current = &((*current)->next);
+  }
+  *current = alarm;
+  
+  return;
+}
+
+void alarmSet(struct alarm* alarm)
+{
+  // Check if the alarm is currently begin suppressed
+  if ((alarm->suppressed) &&
+      (timerHalRun(&(alarm->suppressionTimer)) == HAL_IN_PROGRESS)) {
+    return;
+  } else if (alarm->suppressed) {
+    // If the suppression timer has just expired, unsuppress the alarm now
+    alarm->suppressed = false;
+  }
+
+  alarm->set = true;
+  
+  return;
+}
+
+bool alarmGet(struct alarm* alarm)
+{
+  return alarm->set;
+}
+
+void alarmSupress(struct alarm* alarm)
+{
+  // Clear and suppress the alarm; beginning the suppression timer
+  alarm->suppressed = true;
+  alarm->set = false;
+  timerHalBegin(&(alarm->suppressionTimer),
+                alarm->properties->suppressionTimeout);
+
+  return;
+}
+
+bool alarmCheckAll(struct alarmProperties* properties)
+{
+  struct alarm* current = alarmList;
+  bool allClear = true;
+  
+  // Set the priority to below the "lowest" possible value
+  properties->priority == ALARM_PRIORITY_LOW + 1;
```

Comment:
Where does this get set other than here? Also this line has two equal signs, so it would be a `bool` right?

Reply:
Thanks for catching the typo. Since I havent added the alarms to the other module yet, I guess it might not be clear how it works. Alarms are initialized via `alarmInit` in which they pass in a struct containing their parameters, like their priority, whether or not they need to prevent the watchdog from being reset, etc. The properties struct here is to aggregate the highest priority settings across all alarms for the main loop to manage.